zipline/assets/asset_writer.py

```diff
@@ -268,6 +344,21 @@ def _write_futures(self, futures, bind):
     def _write_equities(self, equities, bind):
         self._write_assets(equities, self.equities, 'equity', bind)
 
+    def check_for_tables(self, engine):
+        """
+        Checks if any tables are present in the current assets database.
+
+        Returns
+        -------
+        bool
+            True if any tables are present, otherwise False.
+        """
+        conn = engine.connect()
+        for table_name in table_names:
```

Comment:
Could it be the case that *all* table names may change, and then this method would return false, even though tables are present? If we don't care about the names of the tables then we could use SQLAlchemy's Inspector to check whether any tables are present:
```python
from sqlalchemy.engine import reflection
...
insp = reflection.Inspector.from_engine(self.eng)
return True if insp.get_table_names() else False
```
http://docs.sqlalchemy.org/en/rel_0_7/core/schema.html?highlight=inspector#sqlalchemy.engine.reflection.Inspector